Remove androidx.work.impl.WorkManagerInitializer from your AndroidManifest.xml

```
> Task :vector:lintVitalFdroidRelease
/home/bubu/projects/oss/fdroiddata/build/im.vector.riotx/vector: Error: Remove androidx.work.impl.WorkManagerInitializer from your AndroidManifest.xml when using on-demand initialization. [RemoveWorkManagerInitializer]

   Explanation for issues of type "RemoveWorkManagerInitializer":
   If an android.app.Application implements
   androidx.work.Configuration.Provider,
   the default androidx.work.impl.WorkManagerInitializer needs to be removed
   from the
   AndroidManifest.xml file.

1 errors, 0 warnings

> Task :vector:lintVitalFdroidRelease FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':vector:lintVitalFdroidRelease'.
> Lint found fatal errors while assembling a release target.
  
  To proceed, either fix the issues identified by lint, or modify your build script as follows:
  ...
  android {
      lintOptions {
          checkReleaseBuilds false
          // Or, if you prefer, you can continue to check for errors in release builds,
          // but continue the build even when errors are found:
          abortOnError false
      }
  }
  ...

```

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -18,6 +18,8 @@ SDK API changes ⚠️:
 
 Build 🧱:
  - Fix lint false-positive about WorkManger (#1012)
+ - Upgrade build-tools from 3.5.3 to 3.6.6
+ - Upgrade gradle from 5.4.1 to 5.6.4
 
 Other changes:
  -

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -17,7 +17,7 @@ SDK API changes ⚠️:
  - 
 
 Build 🧱:
- -
+ - Fix lint false-positive about WorkManger (#1012)
 
 Other changes:
  -

--- a/build.gradle
+++ b/build.gradle
@@ -10,7 +10,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.5.3'
+        classpath 'com.android.tools.build:gradle:3.6.3'
         classpath 'com.google.gms:google-services:4.3.2'
         classpath "com.airbnb.okreplay:gradle-plugin:1.5.0"
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"

--- a/gradle/wrapper/gradle-wrapper.properties
+++ b/gradle/wrapper/gradle-wrapper.properties
@@ -1,6 +1,6 @@
-#Fri Sep 27 10:10:35 CEST 2019
+#Thu Jul 02 12:33:07 CEST 2020
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip

--- a/matrix-sdk-android/src/main/AndroidManifest.xml
+++ b/matrix-sdk-android/src/main/AndroidManifest.xml
@@ -1,5 +1,4 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
     package="im.vector.matrix.android">
 
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
@@ -8,11 +7,6 @@
 
     <application android:networkSecurityConfig="@xml/network_security_config">
 
-        <provider
-            android:name="androidx.work.impl.WorkManagerInitializer"
-            android:authorities="${applicationId}.workmanager-init"
-            android:exported="false"
-            tools:node="remove" />
         <!--
          The SDK offers a secured File provider to access downloaded files.
          Access to these file will be given via the FileService, with a temporary

--- a/vector/build.gradle
+++ b/vector/build.gradle
@@ -106,6 +106,11 @@ def buildNumber = System.env.BUILDKITE_BUILD_NUMBER as Integer ?: 0
 
 android {
     compileSdkVersion 29
+
+    // Due to a bug introduced in Android gradle plugin 3.6.0, we have to specify the ndk version to use
+    // Ref: https://issuetracker.google.com/issues/144111441
+    ndkVersion "21.3.6528147"
+
     defaultConfig {
         applicationId "im.vector.app"
         // Set to API 21: see #405

--- a/vector/build.gradle
+++ b/vector/build.gradle
@@ -232,8 +232,7 @@ android {
     lintOptions {
         lintConfig file("lint.xml")
 
-        // TODO Restore true once pb with WorkManager is fixed
-        abortOnError false
+        abortOnError true
     }
 
     compileOptions {

--- a/vector/src/main/AndroidManifest.xml
+++ b/vector/src/main/AndroidManifest.xml
@@ -247,6 +247,12 @@
 
         <!-- Providers -->
 
+        <!-- Remove WorkManagerInitializer Provider because we are using on-demand initialization of WorkManager-->
+        <provider
+            android:name="androidx.work.impl.WorkManagerInitializer"
+            android:authorities="${applicationId}.workmanager-init"
+            tools:node="remove" />
+
         <provider
             android:name="androidx.core.content.FileProvider"
             android:authorities="${applicationId}.fileProvider"

--- a/vector/src/main/java/im/vector/riotx/VectorApplication.kt
+++ b/vector/src/main/java/im/vector/riotx/VectorApplication.kt
@@ -63,7 +63,13 @@ import java.util.Locale
 import java.util.concurrent.Executors
 import javax.inject.Inject
 
-class VectorApplication : Application(), HasVectorInjector, MatrixConfiguration.Provider, androidx.work.Configuration.Provider {
+import androidx.work.Configuration as WorkConfiguration
+
+class VectorApplication :
+        Application(),
+        HasVectorInjector,
+        MatrixConfiguration.Provider,
+        WorkConfiguration.Provider {
 
     lateinit var appContext: Context
     @Inject lateinit var legacySessionImporter: LegacySessionImporter
@@ -85,6 +91,7 @@ class VectorApplication : Application(), HasVectorInjector, MatrixConfiguration.
     @Inject lateinit var webRtcPeerConnectionManager: WebRtcPeerConnectionManager
 
     lateinit var vectorComponent: VectorComponent
+
     // font thread handler
     private var fontThreadHandler: Handler? = null
 
@@ -157,7 +164,11 @@ class VectorApplication : Application(), HasVectorInjector, MatrixConfiguration.
 
     override fun providesMatrixConfiguration() = MatrixConfiguration(BuildConfig.FLAVOR_DESCRIPTION)
 
-    override fun getWorkManagerConfiguration() = androidx.work.Configuration.Builder().setExecutor(Executors.newCachedThreadPool()).build()
+    override fun getWorkManagerConfiguration(): WorkConfiguration {
+        return WorkConfiguration.Builder()
+                .setExecutor(Executors.newCachedThreadPool())
+                .build()
+    }
 
     override fun injector(): VectorComponent {
         return vectorComponent

--- a/vector/src/main/java/im/vector/riotx/features/form/FormEditTextWithButtonItem.kt
+++ b/vector/src/main/java/im/vector/riotx/features/form/FormEditTextWithButtonItem.kt
@@ -56,6 +56,7 @@ abstract class FormEditTextWithButtonItem : VectorEpoxyModel<FormEditTextWithBut
     }
 
     override fun bind(holder: Holder) {
+        super.bind(holder)
         holder.textInputLayout.isEnabled = enabled
         holder.textInputLayout.hint = hint
 

